src/validation/validateProperties.ts

```diff
@@ -0,0 +1,167 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: MIT
+ */
+
+import {
+    ArrayASTNode,
+    ASTNode,
+    Diagnostic,
+    DiagnosticSeverity,
+    ObjectASTNode,
+    PropertyASTNode,
+    Range,
+    TextDocument,
+} from 'vscode-json-languageservice';
+
+import {
+    findPropChildByName,
+    isArrayNode,
+    isObjectNode,
+} from '../utils/astUtilityFunctions'
+
+import { MESSAGES } from '../constants/diagnosticStrings'
+
+import schema from './validationSchema'
+
+function isObject(obj: any): obj is Object {
+    return obj === Object(obj);
+  }
+
+function getDiagnosticsForNode(
+    rootNode: ASTNode,
+    document: TextDocument,
+    schemaPart: Object,
+    schemaSets: Object
```

Comment:
I was thinking that it might be useful in future when we want to implement fully fledged schema parser but actually maybe it is good to delete it for now. I removed it.